File: app/src/main/java/com/itba/runningMate/services/location/TrackingLocationUpdatesDispatcher.java
package com.itba.runningMate.services.location;

import com.itba.runningMate.services.location.listeners.OnTrackingLocationUpdateListener;
import com.itba.runningMate.services.location.listeners.OnTrackingMetricsUpdateListener;
import com.itba.runningMate.services.location.listeners.OnTrackingUpdateListener;

public interface TrackingLocationUpdatesDispatcher {

    void setOnTrackingUpdateListener(OnTrackingUpdateListener listener);

    void setOnTrackingLocationUpdateListener(OnTrackingLocationUpdateListener listener);

    void setOnTrackingMetricsUpdateListener(OnTrackingMetricsUpdateListener listener);

    void removeTrackingUpdateListener(OnTrackingUpdateListener listener);

    void removeTrackingLocationUpdateListener(OnTrackingLocationUpdateListener listener);

    void removeTrackingMetricsUpdateListener(OnTrackingMetricsUpdateListener listener);

    void callbackLocationUpdate(double latitude, double longitude);

    void callbackDistanceUpdate(float distance);

    void callbackPaceUpdate(long pace);

    void callbackStopWatchUpdate(long elapsedTime);

    boolean areMetricsUpdatesListener();

    boolean areLocationUpdatesListener();

    boolean areUpdatesListener();
}

File: app/src/main/java/com/itba/runningMate/services/location/TrackingLocationUpdatesDispatcher.kt
package com.itba.runningMate.services.location

import com.itba.runningMate.services.location.listeners.OnTrackingLocationUpdateListener
import com.itba.runningMate.services.location.listeners.OnTrackingMetricsUpdateListener
import com.itba.runningMate.services.location.listeners.OnTrackingUpdateListener

interface TrackingLocationUpdatesDispatcher {

    fun setOnTrackingUpdateListener(listener: OnTrackingUpdateListener)

    fun setOnTrackingLocationUpdateListener(listener: OnTrackingLocationUpdateListener)

    fun setOnTrackingMetricsUpdateListener(listener: OnTrackingMetricsUpdateListener)

    fun removeTrackingUpdateListener(listener: OnTrackingUpdateListener)

    fun removeTrackingLocationUpdateListener(listener: OnTrackingLocationUpdateListener)

    fun removeTrackingMetricsUpdateListener(listener: OnTrackingMetricsUpdateListener)

    fun callbackLocationUpdate(latitude: Double, longitude: Double)

    fun callbackDistanceUpdate(distance: Float)

    fun callbackPaceUpdate(pace: Long)

    fun callbackStopWatchUpdate(elapsedTime: Long)

    fun areMetricsUpdatesListener(): Boolean

    fun areLocationUpdatesListener(): Boolean

    fun areUpdatesListener(): Boolean

}